crates/runtime/src/func_env.rs

```diff
@@ -118,17 +117,22 @@ pub struct Inner {
     /// Pointer to `calldata`. Tuple stores `(offset, len)`.
     calldata: Option<(usize, usize)>,
 
-    mode: ProtectedMode,
+    /// The current [`AccessMode`] of the running transaction.
+    mode: AccessMode,
 
+    /// Set of [`Address`] that have been part of at least once `Coins Transfer` during transaction execution.
     touched_accounts: HashSet<Address>,
 }
 
 /// Denotes the capabilities allowed to the executing Account at a given point in time.
 #[derive(Debug, Copy, Clone, PartialEq)]
-pub enum ProtectedMode {
+pub enum AccessMode {
     /// Access to [`AccountStorage`] is not allowed.
     AccessDenied,
 
+    /// Only `Read Access` to [AccountStorage]'s `Immutable Storage` is allowed.
+    ImmutableOnly,
+
     /// Full-Access to [`AccountStorage`] is allowed.
     FullAccess,
```

Comment:
I think we can make it more consistent: `None`, `Read`, `ReadAndWrite`

Reply:
You can still have a method granting access only for reading `Immutable State` and disallowing reading other Sections.

Reply:
Right, that would be `Read` (akin to `ImmutableOnly`), right?

Reply:
Yes